Firestore/core/src/firebase/firestore/util/status.h

```diff
@@ -115,9 +120,28 @@ class ABSL_MUST_USE_RESULT Status {
 
  private:
   static const std::string& empty_string();
+
   struct State {
     State() = default;
     State(const State& other);
+    State(Error code, std::string&& msg);
+
+    struct Deleter {
+      void operator()(const State* ptr) const;
+    };
+    // A `unique_ptr` with a custom deleter. If the pointer's value has been set
+    // to a special value (0x01) to indicate it is moved, invoking the custom
+    // deleter will be a no-op.
+    using StatePtr = std::unique_ptr<State, Deleter>;
+
+    static State* MovedFromIndicator() {
```

Comment:
`MovedFromIndicator` is unused in the header so this can be moved into `status.cc` as a member of the anonymous namespace.

In general try to favor outright hiding over having private methods declared in classes.

Reply:
It is now used by `AssertStatePtr` in header. Also, it returns `State*` which is not accessible from anonymous namespace right?